### ISSUE TITLE
AQL Parsing via Command-Line Tools

### ISSUE THREAD

**Issue by darkfrog26** [arangodb/arangodb#2345]
I'm currently writing an ArangoDB driver for Scala and have finished compile-time AQL interpolation that will fail to compile if the AQL query doesn't validate.  However, at the moment I have to connect to an ArangoDB instance in order to do a compile-time check.  It would be far more useful if there were a command-line tool I could call off to in order to validate the query to avoid requiring a running database just to check for well-formed AQL queries.

**Comment by jsteemann** [comment]
You are right, parsing a query currently requires a running arangod instance.
We have a function for parsing AQL queries there (without executing them), and this function can likely be moved from the arangod server-only part into the common space so that it can be accessed from both arangod and arangosh. Then you could call the parse function from arangosh without having a running server. But it still needs to be checked whether moving the function is easily possible.

**Comment by devgc** [comment]
If possible it would be amazing to have a command line tool for querying a JSON file. Something like: ```ArangoQuery.exe --aql_file file.aql --input_file input.json```. This would be very helpful for a lot of reformatting tasks along with validation.

**Comment by dothebart** [comment]
related to https://github.com/arangodb/arangodb/issues/1755

**Comment by darkfrog26** [comment]
Any updates on this? It's very painful to need a running instance of ArangoDB at compile-time to validate AQL interpolation in my driver.
